### PR TITLE
Changed the Redirect URL

### PR DESCRIPTION
**Pull request by lightshire**
Changed the redirect URL

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -268,10 +272,18 @@ var https = require('https'),
                                 }
                             }
 
+
                             if (response.statusCode === 200) {
+
+                                console.log(self._stream);
```
Comment (ravenjohn) [inline index.js]: way

**Comment by ravenjohn** [comment]
use our jsbeautifyrc

**Comment by lightshire** [comment]
Added
 - set_encoding
 - on(<event_type>, callback);


Would like to request to change options into a json options param
instead of having

```js
        this.secure = false;
        this.started = false;
        this.follow = false;
        this.request_opts = {};
        this.headers = {};
        this.retries = 0;
        this.max_retry = 3;
        this.callbacks = {};
        this.encoding = 'utf8';
```

lets change it soon to

```js
this.options = {
    secure: true,
    ....
    encoding: 'utf8'
};
```

File: index.js [hunk 2]
```diff
@@ -155,6 +162,33 @@ var https = require('https'),
...
+                this.callbacks[event_type] = this.callbacks[event_type].push(next);
+            }
+
+            this.callbacks[event_type] = [next];
+
+            return this;
+        };
+
+        this.emit = function (event_type, err, response) {
+            if (typeof (this.callbacks['data']) !== 'undefined') {
```
Comment (ravenjohn) [inline index.js]: dot notation

File: index.js [hunk 3]
```diff
@@ -155,6 +162,33 @@ var https = require('https'),
...
+                this.callbacks['data'].forEach(function (callback) {
```
Comment (ravenjohn) [inline index.js]: dot notation

File: index.js [hunk 4]
```diff
@@ -155,6 +162,33 @@ var https = require('https'),
...
+                this.callbacks[event_type] = this.callbacks[event_type].push(next);
+            }
+
+            this.callbacks[event_type] = [next];
+
+            return this;
+        };
+
+        this.emit = function (event_type, err, response) {
+            if (typeof (this.callbacks[event_type]) !== 'undefined') {
```
Comment (ravenjohn) [inline index.js]: i think `if (this.callbacks[event_type]) {` is enough

**Comment by lightshire** [comment]
Fixed merge issue.

File: index.js [hunk 5]
```diff
@@ -118,255 +121,325 @@ var https = require('https'),
...
-        this.set_before_json = function(fn) {
+        this.set_before_json = function (fn) {
             this.before_json = fn;
             return this;
         };
 
-        this.follow_redirects = function(max_redirects) {
-            this.max_redirects = +max_redirects || 3;
-            this.follow = true;
+        << << << < HEAD
```
Comment (ravenjohn) [inline index.js]: wat

File: index.js [hunk 6]
```diff
@@ -118,255 +121,325 @@ var https = require('https'),
...
+        this.set_encoding = function (encoding) {
+            this.encoding = encoding;
             return this;
         };
 
-        this.stringify = stringify;
+        this.follow_redirects = function (max_redirects) { === === =
```
Comment (ravenjohn) [inline index.js]: wat

File: index.js [hunk 7]
```diff
@@ -118,255 +121,325 @@ var https = require('https'),
...
+                this.follow_redirects = function (max_redirects) { >>> >>> > d9794801436738f6bd2bcefd7bdd5820610b4e87
```
Comment (ravenjohn) [inline index.js]: wat

File: index.js [hunk 8]
```diff
@@ -118,255 +121,325 @@ var https = require('https'),
...
-                    null,
-                    this,
-                    this.additional_arguments
-                );
-                return this;
-            }
-            logger.log('warn', 'Retrying request');
-            return this.send(this.data);
-        };
+                << << << < HEAD
```
Comment (ravenjohn) [inline index.js]: wat

File: index.js [hunk 9]
```diff
@@ -118,255 +121,325 @@ var https = require('https'),
...
+                    }
 
-            protocol = this.secure ? https : http;
+                    return this;
+                };
 
-            this.request_opts.path = new_path;
-            this.request_opts.method = this.method;
-            this.request_opts.headers = this.headers;
+                this.send = function (data) { === === =
```
Comment (ravenjohn) [inline index.js]: wat

File: index.js [hunk 10]
```diff
@@ -118,255 +121,325 @@ var https = require('https'),
...
+                        this.send = function (data) { >>> >>> > d9794801436738f6bd2bcefd7bdd5820610b4e87
```
Comment (ravenjohn) [inline index.js]: wat

File: index.js [hunk 11]
```diff
@@ -118,255 +121,325 @@ var https = require('https'),
...
 
-                    response.on('end', function() {
-                        var redir,
-                            temp;
+                            this.request_opts.path = new_path;
+                            this.request_opts.method = this.method;
+                            this.request_opts.headers = this.headers;
 
-                        self.response_headers = response.headers;
+                            console.log(this.request_opts);
```
Comment (ravenjohn) [inline index.js]: wat

File: index.js [hunk 12]
```diff
@@ -118,255 +121,325 @@ var https = require('https'),
...
-        };
+                        object.stringify = stringify;
 
-        object.stringify = stringify;
+                        return object;
+                    };
 
-        return object;
-    };
+                module.exports = function (_logger) {
```
Comment (ravenjohn) [inline index.js]: what's the need for indention? it's not part of any code block

File: index.js [hunk 13]
```diff
@@ -154,7 +162,35 @@ var https = require('https'),
             return this.send(this.data);
         };
 
-        this.send = function(data) {
+        << << << < HEAD
```
Comment (ravenjohn) [inline index.js]: wat

**Comment by ravenjohn** [comment]
remove DS store

**Comment by lightshire** [comment]
removed DS_Store and the head.

**Pull request merged**